File: frontend/src/services/entities/request/DangerRequest.ts
import { Danger } from "../Danger";
import { User } from "../User";


export interface DangerRequest {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date | undefined;
    type?: RequestType;
    timestamp?: Date;
    userId?: string;
    dangerId?: string;
    user?: User;
    danger?: Danger;
}

export enum RequestType {
    Create = "Create",
    Resolve = "Resolve"
}

File: backend/DAL/Entities/DangerRequest.cs
namespace DAL.Entities;

public class DangerRequest : Entity {
  public RequestType Type { get; set; }
  public DateTime Timestamp { get; set; }
  public Guid UserId { get; set; }
  public double Lat { get; set; }
  public double Lon { get; set; }
  public string Description { get; set; }
  public Guid CategoryId { get; set; }
  public Guid? DangerId { get; set; }

  // Foreign Keys
  public User User { get; set; }
  public DangerCategory Category { get; set; }
  public Danger? Danger { get; set; }

}

public enum RequestType {
  Create,
  Resolve
}
